sfcasts/ep1/github-service-part-2.md

```diff
@@ -1,103 +1,81 @@
-returns, correct health status for Dino with data sets, sick Dino and a healthy Dino
-are both saying that we have an argument count error, two few arguments to function
-at GitHub service construct, zero pass. And exactly one were expected looking at our,
-uh, service here. That's because we're supposed to be passing in from our test a
-logger instance, and we are not doing that yet. So coming up next, we'll show you how
-to do that. And we'll fix this.
+# GitHub Service - Part 2
 
+Now that we have an idea of what we need the `GithubService` to do, let's add
+the logic inside that we'll use to fetch the issues from the `dino-park` repository
+using GitHub's API.
+
+## Add the client and make a request
+
+In the terminal, add Symfony's HTTP Client to our app with:
+
+```terminal
+composer require symfony/http-client
+```
+
+Which we'll use to call the API. Inside of `GithubService`, instantiate an HTTP 
+client with `$client = HttpClient::create()`. To make a request, the
+`$client->request()` method. We need to tell the client 2 things, 1st: what HTTP
+method to use, like `GET` or `POST`. In this case, the `method` will be `GET`. 
+2nd: the `url` that we'll call is 
+`https://api.github.com/repos/SymfonyCasts/dino-park/issues` will get all the 
+"issues" from the `dino-park` repository that our service needs.
+
+## Parse the HTTP Response
+
+After we call the `request()` method, what happens next? Looking back at the 
+`dino-park` repo, GitHub will return a JSON response that contains the issues we
+see here. Each issue has a title with a dino's name and if the issue has a label 
+attached to it, we'll get that back too. So, in the service, let's set the 
+`HttpResponse` object returned from by the `request()` method on to `$response`.
+
+To filter the issues from the response, add a `foreach()` that loops each
+`$response->toArray()` as an `$issue`. The cool thing about using Symfony's 
+HTTP Client... we don't have to bother transforming the JSON from GitHub into an 
+array - `toArray()` does that heavy lifting for us. Inside this loop, we need to 
+check if the issue title contains the `$dinosaurName`. So 
+`if(str_contains($issue['title'], $dinosaurName))` then we'll `// Do Something`
+with that issue.
+
+I'm going to copy in a private method that will take an array of labels... If 
+one of the labels is a health status label, the method will return the correct
+`HealthStatus` enum based on that label. Now instead of `// Do Something`, we can
+add `$health = $this->getDinoStatusFromLabels()`. And to get the labels for the
+issue, we'll pass in `$issue['labels']`. 
+
+And now we can return `$health`. But... what if an issue doesn't have a health 
+status label? Hmm... at the beginning of this method, set the default `$health` 
+to `HealthStatus::HEALTHY` - because GenLab would *never* forget to put a
+`Sick` label on a dino that isn't feeling well.
+
+Let's check out our tests:
```

Comment:
The "Let's run our tests" or "find your terminal and run your tests" sentences still feel like they're lacking some pizzaz. I'm changing these frequently - e.g. `So... I think we did it! Let's run our tests to be sure.`. That's also a bit better because we're "coming up for air" and giving the user some context. We just did a bunch of work, so let's remind them that "I think we're finished" and then "let's run our test to see".